frontend/server/src/Controllers/Problem.php

```diff
@@ -386,10 +389,17 @@ private static function validateCreateOrUpdate(
      * @omegaup-request-param mixed $update_published
      * @omegaup-request-param mixed $validator
      * @omegaup-request-param mixed $validator_time_limit
-     * @omegaup-request-param mixed $visibility
+     * @omegaup-request-param string $visibility
      */
     public static function apiCreate(\OmegaUp\Request $r): array {
         $r->ensureMainUserIdentity();
+        if (!is_null($r['visibility'])) {
+            $r['intVisibility'] = \OmegaUp\ProblemParams::stringVisibilityToNumeric(
```

Comment:
```suggestion
            $intVisibility = \OmegaUp\ProblemParams::stringVisibilityToNumeric(
```

hay que no modificar `$r`.

Reply:
@lhchavez entonces que modificar la funcion `convertRequestToProblemParams` para que resiva un segundo parametro que sea visibitity?

Reply:
más bien, en `convertRequestToProblemParams()` se puede hacer esa conversión. creo que eso mismo lo mencioné en https://github.com/omegaup/omegaup/pull/4423#discussion_r462267730